ArduPlane/events.cpp

```diff
@@ -146,13 +146,38 @@ void Plane::low_battery_event(void)
     if (flight_stage != AP_SpdHgtControl::FLIGHT_LAND_FINAL &&
         flight_stage != AP_SpdHgtControl::FLIGHT_LAND_PREFLARE &&
         flight_stage != AP_SpdHgtControl::FLIGHT_LAND_APPROACH) {
-    	set_mode(RTL, MODE_REASON_BATTERY_FAILSAFE);
-    	aparm.throttle_cruise.load();
+        set_mode(RTL, MODE_REASON_BATTERY_FAILSAFE);
+        aparm.throttle_cruise.load();
     }
     failsafe.low_battery = true;
     AP_Notify::flags.failsafe_battery = true;
 }
 
+void Plane::critical_battery_event(void)
+{
+    if (failsafe.critical_battery) {
+        return;
+    }
+    gcs_send_text_fmt(MAV_SEVERITY_WARNING, "Critical battery %.2fV used %.0f mAh",
+                      (double)battery.voltage(), (double)battery.current_total_mah());
+    if (flight_stage != AP_SpdHgtControl::FLIGHT_LAND_FINAL &&
+        flight_stage != AP_SpdHgtControl::FLIGHT_LAND_PREFLARE &&
+        flight_stage != AP_SpdHgtControl::FLIGHT_LAND_APPROACH) {
+        set_mode(RTL, MODE_REASON_BATTERY_FAILSAFE);
+        aparm.throttle_cruise.load();
+
+        // Go directly to the landing sequence
+        jump_to_landing_sequence();
```

Comment:
we don't want it to autoland if RTL_AUTOLAND == 0, it should come back home and loiter like a simple RTL.